ui/src/app/applications/components/utils.tsx

```diff
@@ -401,6 +402,107 @@ export function renderResourceMenu(
     );
 }
 
+export function renderResourceButtons(
```

Comment:
There appears to be some level of duplicate code between the two functions `renderResourceButtons` and the current `renderResourceMenu`. Perhaps try to pull out common code into non-exported functions, if possible?

Reply:
100% agree. We should try to reuse the existing `renderResourceMenu` function.

Reply:
Update: Regina, I think you refactored this, so this specific comment can be resolved.